x/mongo/driver/topology/server.go

```diff
@@ -653,12 +657,14 @@ func (s *Server) check() (description.Server, error) {
 		heartbeatConn := initConnection{s.conn}
 		baseOperation := s.createBaseOperation(heartbeatConn)
 		previousDescription := s.Description()
+		streamable := previousDescription.TopologyVersion != nil
 
 		switch {
 		case s.conn.getCurrentlyStreaming():
+
```

Comment:
The `duration` variable needs to be set in this block. Per the SDAM monitoring spec, when streaming responses, the duration should account for the time spent waiting for the server response. This means that the duration isn't an accurate representation of server RTT in this case. I think we can simplify a bit by avoid the duplicated time calculations:

```
if descPtr == nil && err == nil {
   // using existing connection
   
   // setup stuff...
   
   start = time.Now()
   switch {
       case s.conn.getCurrentlyStreaming():
       case streamable:
       default:
   }
   duration = time.Since(start)
}
```

None of these cases do any blocking work besides the isMaster execution, so it should be fine to measure the time this way.